### PR TITLE
fix: map viewer functionality MA-897

### PR DESCRIPTION
**Pull request by e0**

### CODE REVIEW

**Review by mihai-sysbio — APPROVED** [review]
Excellent work! 
Btw I'm not sure what changed, but the selection is now (again) broken on my machine. There's no need to do something about it, but maybe you have an idea about what was recently changed.

**Comment by mihai-sysbio** [comment]
I don't see anything that is worth stopping the merging of this PR, so I'm going to merge it, but let's continue the discussion.

**Comment by e0** [comment]
@mihai-sysbio Thanks for the review. My email notifications stopped working for some reason this afternoon, just saw the Jira notification now so sorry for not responding earlier.

File: frontend/src/components/explorer/mapViewer/MapLoader.vue [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
 export default {
   name: 'MapLoader',
-  props: {
-    loading: {
-      type: Boolean,
-      required: true,
-    },
+  computed: {
+    ...mapState({
+      loading: state => state.maps.loading,
```
[the remark this answers is not in the document]
Comment (e0) [reply frontend/src/components/explorer/mapViewer/MapLoader.vue:14]: I agree that parent-child props are good for certain cases. I felt like in this specific case the loading status is used in several different components and it was easier to keep everything in sync with using it in a state. What would be the argument for leaving it as a prop?

**Comment by mihai-sysbio** [comment]
After delaying the review I was in a bit of a rush to merge it - sorry.
The fixes I pushed have nothing to do with the checklist in MA-897, they were just a few minor things I thought I could slide in next to your work.

File: frontend/src/components/explorer/mapViewer/MapLoader.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mihai-sysbio) [reply frontend/src/components/explorer/mapViewer/MapLoader.vue:14]: I think I would leave everything as props where possible - it feels more lightweight than going through the state.
In this particular case, I believe the state was needed, but I still wanted to bring this up.

File: frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue [hunk 2]
```diff
@@ -206,10 +206,14 @@ export default {
 };
 </script>
 
-<style lang="scss">
+<style lang="scss" scoped>
   #sidebar_mapviewer {
```
[the remark this answers is not in the document]
Comment (e0) [reply frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue:209]: Sorry, I think this was an oversight. While I do think it's good practice to have `scoped` for all components it was not my intention to hint it by changing just one file. I think this was probably due to the fact that I added some extra css outside the id block and forgot to remove the `scoped` tag after removing what I was experimenting with.

File: frontend/src/store/modules/models.js [hunk 3]
```diff
@@ -27,14 +27,16 @@ const getters = {
...
+      const models = await modelsApi.fetchModels();
+      commit('setModelList', models.sort((a, b) => (a.short_name.toLowerCase() < b.short_name.toLowerCase() ? -1 : 1)));
+    }
   },
   /* eslint-disable no-shadow */
-  async selectModel({ dispatch, commit, getters, state }, modelShortName) {
+  async selectModel({ dispatch, commit, getters }, modelShortName) {
     if (state.modelList.length === 0) {
-      await dispatch('getModels');
+    await dispatch('getModels');
```
Comment (e0) [inline frontend/src/store/modules/models.js:39]: Not on the right computer right now so I haven't tested this. But how come the state can be omitted on line 37? Also the indentation seems off now on line 39.

File: frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihai-sysbio) [reply frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue:209]: I've added a checklist item for `scoped` under MA-718.

File: frontend/src/components/explorer/mapViewer/MapLoader.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (e0) [reply frontend/src/components/explorer/mapViewer/MapLoader.vue:14]: Ah I see. I think my intuition is that I prefer vuex for when something is used in more than two components. I'm not sure if this is reflected in the code though. Definitely a good discussion to have and something we should add to code guidelines.

File: frontend/src/store/modules/models.js (hunk 3 above)
Comment (mihai-sysbio) [reply frontend/src/store/modules/models.js:39]: Good catch! It's a mistake - I had moved the condition check inside `getModels` above. I'll fix this on another branch.

**Comment by e0** [comment]
> Excellent work!
> Btw I'm not sure what changed, by the selection is now (again) broken on my machine. There's no need to do something about it, but maybe you have an idea about what was recently changed.

Do you remember at which commit it was working for you?

**Comment by mihai-sysbio** [comment]
Maybe when the 3D network viewer was at `0.1.6` or `0.1.7`, I'm not sure.

**Comment by e0** [comment]
Ok, it would be difficult for me to identify the possible cause without more info since it has always worked for me.

**Pull request merged**